### PR TITLE
feat(web-analytics): Add custom bounce rate control

### PR DESCRIPTION
**Pull request by robbie-c**
## Problem

See https://posthog.slack.com/archives/C05LJK1N3CP/p1734713816519279 for context

Docs https://github.com/PostHog/posthog.com/pull/10206

## Changes

Make the bounce rate duration threshold configurable

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Added a test - see diff

### CODE REVIEW

File: frontend/src/scenes/settings/environment/BounceRateDuration.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        currentTeam?.modifiers?.bounceRateDurationSeconds ?? currentTeam?.default_modifiers?.bounceRateDurationSeconds
+    const [bounceRateDuration, setBounceRateDuration] = useState<number | undefined>(savedDuration)
+
+    const handleChange = (duration: number | undefined): void => {
+        updateCurrentTeam({ modifiers: { ...currentTeam?.modifiers, bounceRateDurationSeconds: duration } })
+    }
+
+    return (
+        <>
+            <p>Choose how long a user can stay on a page, before the session is not a bounce.</p>
```
Comment (rafaeelaudibert) [inline frontend/src/scenes/settings/environment/BounceRateDuration.tsx:21]: Let's make it clear we're talking seconds here

File: posthog/hogql/database/schema/sessions_v1.py [hunk 2]
```diff
@@ -230,7 +236,10 @@ def arg_max_merge_field(field_name: str) -> ast.Call:
                             # if session duration >= 10 seconds, not a bounce
```
Comment (rafaeelaudibert) [inline posthog/hogql/database/schema/sessions_v1.py:236]: Let's update the comment

Btw, we were using 10 seconds, not 30, lol, website docs are probably outdated

File: posthog/hogql/database/schema/sessions_v2.py [hunk 3]
```diff
@@ -37,6 +37,9 @@
 if TYPE_CHECKING:
     from posthog.models.team import Team
 
+
+DEFAULT_BOUNCE_RATE_DURATION_SECONDS = 30
```
Comment (rafaeelaudibert) [inline posthog/hogql/database/schema/sessions_v2.py:41]: Default was actually 10 if you look at the old line 274

File: posthog/hogql/database/schema/sessions_v2.py [hunk 4]
```diff
@@ -271,7 +279,10 @@ def arg_max_merge_field(field_name: str) -> ast.Call:
                                 # if session duration >= 10 seconds, not a bounce
```
Comment (rafaeelaudibert) [inline posthog/hogql/database/schema/sessions_v2.py:279]: There are a bunch of comments that need to be updated

File: posthog/hogql/database/schema/sessions_v2.py (hunk 3 above)
Comment (robbie-c) [reply posthog/hogql/database/schema/sessions_v2.py:41]: I should not try to work on a train 🙃

File: posthog/hogql/database/schema/sessions_v1.py (hunk 2 above)
Comment (robbie-c) [reply posthog/hogql/database/schema/sessions_v1.py]: https://github.com/PostHog/posthog.com/pull/10206

File: frontend/src/scenes/settings/environment/BounceRateDuration.tsx (hunk 1 above)
Comment (robbie-c) [reply frontend/src/scenes/settings/environment/BounceRateDuration.tsx]: Done

**Review by rafaeelaudibert — APPROVED** [review]
LGTM!

Is this `query_modifier` a JSON object for each team? I'm asking because I don't see a DB migration

**Pull request merged**